[C10D] Refactor NCCL logs to use common prefix helper

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #115807
* #115803
* #115801
* __->__ #115800
* #115798
* #115771

Put the repeated code that string formats [Rank {rank}] in one place.

Sets up for the next PR that also adds more info to this prefix.

(Does not change exception messages, which could be done as well.
Exception messages are not formatted quite the same way. Tries
instead to keep from changing log behavior (in this PR) and only
refactor code.

Did limited testing (some logs were observed OK).

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @yf225

wconstab committed:
[C10D] Refactor NCCL logs to use common prefix helper

Put the repeated code that string formats [Rank {rank}] in one place.

Sets up for the next PR that also adds more info to this prefix.

(Does not change exception messages, which could be done as well.
Exception messages are not formatted quite the same way. Tries
instead to keep from changing log behavior (in this PR) and only
refactor code.

Did limited testing (some logs were observed OK).

[ghstack-poisoned]

XilunWu reviewed (commented):
Unifying the log head makes it easier to search for a specific rank's log. I think this change is good. Want to see others' opinion.

fduwjj reviewed (approved):
LGTM

wconstab commented:
@pytorchbot merge -i

wconstab committed:
Update on "[C10D] Refactor NCCL logs to use common prefix helper"


Put the repeated code that string formats [Rank {rank}] in one place.

Sets up for the next PR that also adds more info to this prefix.

(Does not change exception messages, which could be done as well.
Exception messages are not formatted quite the same way. Tries
instead to keep from changing log behavior (in this PR) and only
refactor code.

Did limited testing (some logs were observed OK).

[ghstack-poisoned]
